subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -1,94 +1,502 @@
+			hp->tx_pwr = 1;
+			pp++;
+		} else {
+			hs->tx_pwr = 1;
 		}
 	}
+	if (_hs.tx_pwr) {
+		_ps++;
+
+		hs->tx_pwr = 1;
+	}
+	if (hs->tx_pwr) {
+		ps++;
+	}
+
+	/* No ACAD in Primary channel PDU */
+	/* TODO: ACAD in Secondary channel PDU */
+
+	/* Calc primary PDU len */
+	_pri_len = _pp - (u8_t *)_p;
+	pri_len = pp - (u8_t *)p;
+	p->ext_hdr_len = pri_len - offsetof(struct pdu_adv_com_ext_adv,
+					    ext_hdr_adi_adv_data);
+
+	/* set the primary PDU len */
+	pri->len = pri_len;
+
+	/* Calc secondary PDU len */
+	_sec_len = _ps - (u8_t *)_s;
+	sec_len = ps - (u8_t *)s;
+	s->ext_hdr_len = sec_len - offsetof(struct pdu_adv_com_ext_adv,
+					    ext_hdr_adi_adv_data);
+
+	/* TODO: Check AdvData overflow */
+
+	/* Fill AdvData in secondary PDU */
+	memcpy(ps, data, len);
+
+	/* set the secondary PDU len */
+	sec->len = sec_len + len;
+
+	/* Start filling primary PDU extended header  based on flags */
+
+	/* No AdvData in primary channel PDU */
+
+	/* No ACAD in primary channel PDU */
+
+	/* Tx Power */
+	if (hp->tx_pwr) {
+		*--pp = *--_pp;
+	} else if (hs->tx_pwr) {
+		*--ps = *--_ps;
+	}
+
+	/* No SyncInfo in primary channel PDU */
+	/* SyncInfo in secondary channel PDU */
+	if (hs->sync_info) {
+		_ps -= sizeof(struct ext_adv_sync_info);
+		ps -= sizeof(struct ext_adv_sync_info);
+
+		memcpy(ps, _ps, sizeof(struct ext_adv_sync_info));
+	}
+
+	/* AuxPtr */
+	if (_hp.aux_ptr) {
+		_pp -= sizeof(struct ext_adv_aux_ptr);
+	}
+	{
+		struct ext_adv_aux_ptr *aux;
+
+		pp -= sizeof(struct ext_adv_aux_ptr);
+
+		/* NOTE: Aux Offset will be set in advertiser LLL event */
+		aux = (void *)pp;
+		aux->chan_idx = 0; /* FIXME: implementation defined */
+		aux->ca = 0; /* FIXME: implementation defined */
+		aux->offs_units = 0; /* FIXME: implementation defined */
+		aux->phy = find_lsb_set(lll->phy_s) - 1;
+	}
+	if (_hs.aux_ptr) {
+		struct ext_adv_aux_ptr *aux;
+
+		_ps -= sizeof(struct ext_adv_aux_ptr);
+		ps -= sizeof(struct ext_adv_aux_ptr);
+
+		/* NOTE: Aux Offset will be set in advertiser LLL event */
+		aux = (void *)ps;
+		aux->chan_idx = 0; /* FIXME: implementation defined */
+		aux->ca = 0; /* FIXME: implementation defined */
+		aux->offs_units = 0; /* FIXME: implementation defined */
+		aux->phy = find_lsb_set(lll->phy_s) - 1;
+	}
 
-	lll_adv_data_enqueue(&adv->lll, idx);
+	/* ADI */
+	{
+		struct ext_adv_adi *ap, *as;
+		u16_t did = UINT16_MAX;
+
+		pp -= sizeof(struct ext_adv_adi);
+		ps -= sizeof(struct ext_adv_adi);
+
+		ap = (void *)pp;
+		as = (void *)ps;
+
+		if (_hp.adi) {
+			struct ext_adv_adi *_adi;
+
+			_pp -= sizeof(struct ext_adv_adi);
+			_ps -= sizeof(struct ext_adv_adi);
+
+			/* NOTE: memcpy shall handle overlapping buffers */
+			memcpy(pp, _pp, sizeof(struct ext_adv_adi));
+			memcpy(ps, _ps, sizeof(struct ext_adv_adi));
+
+			_adi = (void *)_pp;
+			did = _adi->did;
+		} else {
+			ap->sid = adv->sid;
+			as->sid = adv->sid;
+		}
+
+		if ((op == 0x04) || len || (_pri_len != pri_len)) {
+			did++;
+		}
+
+		ap->did = did;
```

Comment:
As far as I can tell, this should be okay without endianness conversion. However, it would be preferable to convert endianness anyway to be able to have identical traces between LE and BE architectures.

Reply:
good catch. Fixed.